micro/disj.go

```diff
@@ -11,15 +11,7 @@ scheme code:
 		)
 	)
 */
-func DisjointO(gs ...Goal) Goal {
-	if len(gs) == 0 {
-		return FailureO
-	}
-	if len(gs) == 1 {
-		return gs[0]
-	}
-	g1 := gs[0]
-	g2 := DisjointO(gs[1:]...)
+func DisjointO(g1, g2 Goal) Goal {
```

Comment:
Why make the new function concurrentDisjPlus have elipses, but remove them here?
`func concurrentDisjPlus(gs ...micro.Goal) micro.Goa;`

Reply:
Whichever way we go, lets keep the symmetry

Reply:
Theres a difference between what the paper calls `disj` and `conj`, which only take two goals, and macros `disj+` and `conj+` which extend these to arbitrary number of goals.
You had a DisjointO and ConjunctionO, which were somewhere in between.
They were variadic but did not include an automatic Zzz wrapper like in the paper.
I fixed DisjointO and ConjunctionO (please lets rename these) to match the elementary `disj` and `conj`.